coalib/bearlib/naming_conventions/__init__.py

```diff
@@ -130,3 +130,45 @@ def to_spacecase(string):
     return re.sub('(?<=[^\s])([A-Z])',
                   lambda match: ' ' + match.group(1),
                   string)
+
+
+def to_kebabcase(string):
+    """
+    Converts the given string to kebab-case.
+
+    >>> to_kebabcase('HelloWorld')
+    'hello-world'
+    >>> to_kebabcase('__Init__File__')
+    '--init-file--'
+    >>> to_kebabcase('')
+    ''
+    >>> to_kebabcase('already-kebab-case')
+    'already-kebab-case'
+    >>> to_kebabcase('   string  ')
+    '---string--'
+    >>> to_kebabcase('ABCde.F.G..H..IH')
+    'a-b-cde.f.g..h..i-h'
```

Comment:
I agree with @gs0510 on the above point . I've used it in lisp and lisp's kebab case does not begin with a hyphen. But instead of multiple hyphens , its preferred to strip white-lines from both ends and it contains blocks of lower case words separated by single hyphen. It's better if you implement that.